Makefile

```diff
@@ -7,8 +7,13 @@ else
     CFLAGS += -O2
 endif
 
-mkmimo: mkmimo.o
+ifdef MULTITHREADED
+mkmimo: mkmimo_multithreaded.o queue.o
 	$(CC) -o $@ $(LDFLAGS) $^
+else
+mkmimo: main.o utils.o mkmimo_nonblocking.o
```

Comment:
Yeah, looks good to me.

main.o may be missing for the `MULTITHREADED` case, but otherwise I think the build rule seems fine.